nornir_ansible/plugins/inventory/ansible.py

```diff
@@ -178,10 +181,17 @@ def _vars_file_exists(path: str) -> bool:
             bool: True if a file exists with of of the extension
 
         """
-        for ext in VARS_FILENAME_EXTENSIONS:
-            if Path(f"{path}{ext}").is_file():
-                return True
-        return False
+        if not Path(f"{path}{host}").is_dir():
+          exts = '|'.join(map(str,VARS_FILENAME_EXTENSIONS))
+          pattern = re.compile(r'.*{}({})$'.format(re.escape(host), exts), re.IGNORECASE)
+          found_files = [f for f in os.listdir(path) if pattern.search(f)]
+          if len(found_files) > 1:
+            raise Exception("More than one file found")
+          if len(found_files) == 1:
+            if Path(f"{path}{found_files[0]}").is_file():
+              return found_files[0]
+          else:
+            return None
```

Comment:
cant return None if we are hinting the return type is as string here. I imagine can just return empty string though since that will be false where this is called.